proposals/NativeInt.md

```diff
@@ -0,0 +1,345 @@
+# Native-sized integers
+
+## Summary
+[summary]: #summary
+
+Language support for a native-sized signed and unsigned integer types.
+
+The motivation is for interop scenarios and for low-level libraries.
+
+## Design
+[design]: #design
+
+The identifiers `nint` and `nuint` are new contextual keywords that represent native signed and unsigned integer types.
+The identifiers are only treated as keywords when name lookup does not find a viable result at that program location.
+```C#
+nint x = 3;
+var y = nameof(nuint);
+var z = nint.Zero;
+```
+
+The types `nint` and `nuint` are represented by the underlying types `System.IntPtr` and `System.UIntPtr` with compiler surfacing additional conversions and operations for those types as native ints.
+
+### Constants
+
+There is no direct syntax for native int literals. Explicit casts of other integral constant values can be used instead: `(nint)42`.
+
+`nint` constants are in the range [ `int.MinValue`, `int.MaxValue` ].
+
+`nuint` constants are in the range [ `uint.MinValue`, `uint.MaxValue` ].
+
+There are no `MinValue` or `MaxValue` fields on `nint` or `nuint` because, other than `nuint.MinValue`, those values cannot be emitted as constants.
+
+Constant folding is supported for all unary operators { `+`, `-`, `~` } and binary operators { `+`, `-`, `*`, `/`, `%`, `==`, `!=`, `<`, `<=`, `>`, `>=`, `&`, `|`, `^`, `<<`, `>>` }.
+Constant folding operations are evaluated with `Int32` and `UInt32` operands rather than native ints for consistent behavior regardless of compiler platform.
+
+### Conversions
+There are identity conversions between native ints and the underlying types in both directions.
```

Comment:
>between native ints and the underlying types [](start = 31, length = 44)

Please make these explicit.  Specifically, please say that "There is an identity conversion between nint and IntPtr, and between nuint and UIntPtr."